draft-sharabayko-srt.md

```diff
@@ -1667,8 +1690,96 @@ of the higher bit of the difference.
+- HS Extensions: none.
+
+Legacy HS Version 4 clients do not look at the HS Version value,
+whereas HS Version 5 clients can detect version 5.
+The parties only continue with the HS Version 5 Rendezvous process when Version is set to 5
+for both. Otherwise the process continues exclusively according to Version 4 rules {{GHG04b}}.
+Implementations MUST support HS Version 5, and MAY not support HS Version 4.
+
+The WAVEAHAND Handshake packet SHOULD not have extensions.
+
+With SRT Handshake Version 5 Rendezvous, both parties create a cookie for a process called the
+"cookie contest". This is necessary for the assignment of Initiator and Responder
+roles. Each party generates a cookie value (a 32-bit number) based on the host,
+port, and current time with 1 minute accuracy. This value is scrambled using
+an MD5 sum calculation. The cookie values are then compared with one another.
+
+Since it is impossible to have two sockets on the same machine bound to the same NIC
+and port and operating independently, it is virtually impossible that the
+parties will generate identical cookies. However, this situation may occur if an
+application tries to "connect to itself" - that is, either connects to a local
+IP address, when the socket is bound to INADDR_ANY, or to the same IP address to
+which the socket was bound. If the cookies are identical (for any reason), the
+connection will not be made until new, unique cookies are generated (after a
+delay of up to one minute). In the case of an application "connecting to itself",
+the cookies will always be identical, and so the connection will never be established.
+
+If there is no response from a peer the WAVEAHAND handshake SHOULD be repeated every 250 ms
+until a connection timeout expires. The connection timeout value is defined by the implementation.
+
+If a WAVEAHAND of CONCLUSION handshake is received from the peer, the state is transitioned to the Attention.
+
+#### Conclusion {#conclusion-state}
+
+In the Conclusion state the party has received and now knows the peer's cookie value.
+Thus it can perform the Cookie Contest
+(compare both cookie values according to {{cookie-contest}})
+and resolve its role.
+The resolution of the Handshake Role (Initiator or Responder) is essential for further processing.
+
+Initiator replies with a Conclusion request handshake:
+
+- HS Version: 5
+- Extension field: appropriate flags.
+- Encryption field: advertised PBKEYLEN
+- Handshake Type: CONCLUSION.
+- Required Handshake Extension: HS Extension Message ({{handshake-extension-msg}})
+  with HS Extension Type SRT_CMD_HSREQ.
+- Other handshake extensions are allowed.
+
+If encryption is on, the Initiator will use either his
+own cipher family and block size or the one received from Alice (if she has advertised
+those values).
+
+The Responder with a Conclusion or a WAVEAHAND handshake without extensions until it receives the Conclusion Request from the peer:
```

Comment:
This sentence is unclear and does not seem to sensibly follow from preceding lines 1753-1755.